Libs/MRML/Core/vtkMRMLPlotLayoutNode.h

```diff
@@ -0,0 +1,261 @@
+/*==============================================================================
+
+  Copyright (c) Kapteyn Astronomical Institute
+  University of Groningen, Groningen, Netherlands. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Davide Punzo, Kapteyn Astronomical Institute,
+  and was supported through the European Research Consil grant nr. 291531.
+
+==============================================================================*/
+
+#ifndef __vtkMRMLPlotLayoutNode_h
+#define __vtkMRMLPlotLayoutNode_h
+
+#include "vtkMRMLNode.h"
+
+class vtkCollection;
+class vtkDataObject;
+class vtkMRMLPlotNode;
+class vtkStringArray;
+
+class PlotIDMap;
+class PlotLayoutPropertyMap;
+
+#include <string>
+
+/// \brief MRML node for referencing a collection of data to plot.
+class VTK_MRML_EXPORT vtkMRMLPlotLayoutNode : public vtkMRMLNode
+{
+ public:
+  //----------------------------------------------------------------
+  /// Standard methods for MRML nodes
+  //----------------------------------------------------------------
+
+  static vtkMRMLPlotLayoutNode *New();
+  vtkTypeMacro(vtkMRMLPlotLayoutNode,vtkMRMLNode);
+
+  void PrintSelf(ostream& os, vtkIndent indent);
+
+  virtual vtkMRMLNode* CreateNodeInstance();
+
+  ///
+  /// Set node attributes
+  virtual void ReadXMLAttributes( const char** atts);
+
+  ///
+  /// Write this node's information to a MRML file in XML format.
+  virtual void WriteXML(ostream& of, int indent);
+
+  ///
+  /// Copy the node's attributes to this object
+  virtual void Copy(vtkMRMLNode *node);
+
+  ///
+  /// Get node XML tag name (like Volume, Model)
+  virtual const char* GetNodeTagName() {return "PlotLayout";};
+
+  ///
+  /// Method to propagate events generated in mrml
+  virtual void ProcessMRMLEvents ( vtkObject *caller, unsigned long event, void *callData );
+
+  //----------------------------------------------------------------
+  /// Access methods
+  //----------------------------------------------------------------
+
+  ///
+  /// Add and Observe a Plot to the PlotLayout. Parameter "name" is used for
+  /// referencing the Plot when setting properties for plotting the
+  /// Plot or for removing the Plot from the PlotLayout.
+  virtual void AddAndObservePlot(const char *name, const char * id);
```

Comment:
I wrote the API lookig at the ChartNode where the the plot ID are saved in a map containing both the name and the ID of the plot. I'll see if the name can be taken out and store only the ID. The issue is that in the controller widget the Plot are identified by name. I'll revise it and we can review this at the new iteration.